### PR TITLE
Document periodic cache cleanup in Release Notes

### LINKED ISSUE #5800: Document cache cleanup in Release Notes
Document cleanup of shared and version-specific caches.

### PR DESCRIPTION
**Pull request by marcphilipp**
Resolves #5800.

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -14,6 +14,14 @@ Add-->
 
 Starting with this release, it is now possible to use SNAPSHOT plugin versions in the `plugins {}` and `pluginManagement {}` blocks.
 
+### Periodic cache cleanup
+
+Caching has always been one of the strong suits of Gradle. Over time, more and more persistent caches have been added to improve performance and support new features, requiring more and more disk space on build servers and developer workstations. Gradle now addresses one of the most highly voted issues on GitHub and introduces the following cleanup strategies:
+
+- Version-specific cache directories in `GRADLE_USER_HOME/caches/<gradle-version>/` are checked periodically (at most every 24 hours) for whether they are still in use. If not, directories for release versions are deleted after 30 days of inactivity, snapshot versions after 7 days of inactivity. Moreover, the corresponding Gradle distributions at `GRADLE_USER_HOME/wrapper/dists/gradle-<gradle-version>-{bin,all}/` are deleted as well, if present.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/release/notes.md]: Reading this makes me wonder if we are covered when users have their own custom Gradle distribution.

**Comment by oehme** [comment]
Is this also documented in the user guide? Release notes are not read by many.

**Comment by marcphilipp** [comment]
@oehme Where would we put in the User Guide? In the [Dependency Cache](https://docs.gradle.org/current/userguide/dependency_cache.html) chapter?

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (marcphilipp) [reply subprojects/docs/src/docs/release/notes.md]: Right, if they use a different naming pattern, we're not going to find and delete the distribution.

**Comment by oehme** [comment]
Yeah I think so

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (marcphilipp) [reply subprojects/docs/src/docs/release/notes.md]: This PR adds support for custom distributions: #5906.

**Comment by marcphilipp** [comment]
I'll update the User Guide in #5917.

**Pull request merged**